Formula/gap.rb

```diff
@@ -11,46 +19,141 @@ class Gap < Formula
+      "xmodalg", "yangbaxter",
+    ]
+
+    # make doc and test targets, I don't actually call them
+    makefile_packages = [
+      "4ti2interface", "autodoc", "cap", "examplesforhomalg",
+      "gaussforhomalg", "generalizedmorphismsforcap", "gradedmodules",
+      "gradedringforhomalg", "homalg", "homalgtocas", "io_forhomalg",
+      "linearalgebraforcap", "localizeringforhomalg",
+      "matricesforhomalg", "modulepresentationsforcap", "modules",
+      "monoidalcategories", "nconvex", "ringsforhomalg", "sco",
+      "toolsforhomalg", "toricvarieties",
+    ]
+
+    # These two packages either don't build or don't work
+    # "cddinterface", "xgap",
+    configure_packages = [
+      "anupq", "caratinterface", "crypting", "curlinterface", "cvec",
+      "datastructures", "deepthought", "digraphs", "ferret", "float",
+      "gauss", "grape", "io", "json", "normalizinterface", "nq",
+      "orb", "profiling", "semigroups", "simpcomp", "zeromqinterface",
+      ]
+
+    old_configure_packages = [
+      "ace", "browse", "cohomolo", "edim", "example", "fplsa",
+      "guava", "kbmag",
+    ]
+
+    # These package have autogen.sh available
+    # I don't think there is a need to run it, but we could if we wanted
+    autogen_packages = [
+      "anupq", "cddinterface", "curlinterface", "digraphs", "ferret",
+      "float", "guava", "io", "normalizinterface", "nq", "semigroups",
+      "simpcomp", "xgap", "zeromqinterface",
+    ]
+
+    # Run special commands after installation
+    special_packages = {
+      # Even with x11 installed, it doesn't seem to work
+      # "xgap" => "cp bin/xgap.sh $GAPROOT/bin/xgap.sh",
+    }
+
+
+    # Start actually building GAP
+    if build.head?
+      system "./autogen.sh"
+    end
+
+    libexec.install Dir["pkg"]
+
+    system "./configure", "--prefix", libexec/"gap", "--with-readline=#{Formula["readline"].opt_prefix}"
+    if build.head?
+      system "make", "bootstrap-pkg-full"
+      # system "make", "doc"      # Do we need this?
     end
+    system "make", "install"
 
     # Create a symlink `bin/gap` from the `gap` binary
-    bin.install_symlink libexec/"gap" => "gap"
+    bin.install_symlink libexec/"gap/bin/gap" => "gap"
 
     ohai "Building included packages. Please be patient, it may take a while"
+    pkg_dir = "#{libexec}/pkg"
+
     cd libexec/"pkg" do
-      # NOTE: This script will build most of the packages that require
-      # compilation. It is known to produce a number of warnings and
-      # error messages, possibly failing to build several packages.
-      system "../bin/BuildPackages.sh", "--with-gaproot=#{libexec}"
+
+      system "mkdir", "#{libexec}/gap/lib/gap/pkg/"
+
+      # The makefiles appear to only be used for docs...
+      # The BuildPackages.sh script didn't call them
+      no_compilation_packages.concat(makefile_packages).each do |pkg|
+        system "cp", "-R", pkg, "#{libexec}/gap/lib/gap/pkg/"
+      end
+
+      prerequisites_packages.each do |pkg|
```

Comment:
Instead of maintaining a `prerequisites_packages` list, why not just invoke `prerequisites.sh` in all directories where it is present?